src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -20,7 +22,8 @@ class Crystal::Doc::Generator
       types.insert 0, program_type
     end
 
-    generate_docs program_type, types
+    generate_docs program_type, types if @format == "html"
+    File.write("#{@dir}/docs.json", types.to_json) if @format == "json"
```

Comment:
A `if/else` might make it more obvious what's going on. Also I'd say extract the second line into a method to keep the same level of abstraction in this method.